File: median-of-two-sorted-arrays.java
import java.util.Arrays;

class Solution
{
    public double findMedianSortedArrays(int[] nums1, int[] nums2)
    {
        int[] all_nums = new int[nums1.length + nums2.length];
        
        System.arraycopy(nums1, 0, all_nums, 0, nums1.length);
        System.arraycopy(nums2, 0, all_nums, nums1.length, nums2.length);
        Arrays.sort(all_nums);

        return all_nums.length % 2 == 0 ? ((all_nums[((all_nums.length)/2)]) + (all_nums[((all_nums.length)/2) - 1])) / 2.0 : all_nums[((all_nums.length-1)/2)];
    }
}

File: python/hard/median-of-two-sorted-arrays.py
# hard (https://leetcode.com/problems/median-of-two-sorted-arrays/)


from typing import List


class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        all_nums = [*nums1, *nums2]
        all_nums.sort()

        return ((all_nums[((len(all_nums))//2)]) + (all_nums[((len(all_nums))//2) - 1])) / 2.0 if len(all_nums) % 2 == 0 else all_nums[((len(all_nums)-1)//2)]